meta-resin-common/recipes-containers/docker-disk/docker-resin-supervisor-disk/start-resin-supervisor

```diff
@@ -7,6 +7,14 @@ SUPERVISOR_IMAGE_ID=$(balena inspect --format='{{.Id}}' "$SUPERVISOR_IMAGE:$SUPE
 SUPERVISOR_CONTAINER_IMAGE_ID=$(balena inspect --format='{{.Image}}' resin_supervisor || echo "")
 
 runSupervisor() {
+    # If self-signed root CA exists, add the CA for the specified registry
+    BALENA_DECODED_ROOT_CA=$(if [ ! -z "${BALENA_ROOT_CA}" ]; then echo "${BALENA_ROOT_CA}" | base64 -d -w0; fi)
```

Comment:
This looks a little bit strange. Basically, you define a variable (based on an if) and afterwards you `if` the generated variable. I think it would be cleaner something like:

```
if [ ! -z "${BALENA_ROOT_CA}" ]; then
    BALENA_DECODED_ROOT_CA="$(echo "${BALENA_ROOT_CA}" | base64 -d -w0)"
    CERT_DIR="/mnt/state/root-overlay/etc/docker/certs.d/${REGISTRY_ENDPOINT}"
    mkdir -p "${CERT_DIR}"
    echo "${BALENA_DECODED_ROOT_CA}" > "${CERT_DIR}/ca.crt"
fi
```

As well, this piece of code doesn't run at every boot. It runs only if the supervisor container doesn't exist. So if you write the certificate, start the container and by chance, the certificate is not synced to the disk but the container is, you might end up at next boot with a corrupted certificate. So what I recommend is either to make sure the certificate file hits the disk (atomic, durable) or always generate the file before starting the container or creating a new container.